sdk/azidentity/internal/customtokenendpoint/policy.go

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package customtokenendpoint
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+const (
+	AzureKubernetesCAData        = "AZURE_KUBERNETES_CA_DATA"
+	AzureKubernetesCAFile        = "AZURE_KUBERNETES_CA_FILE"
+	AzureKubernetesSNIName       = "AZURE_KUBERNETES_SNI_NAME"
+	AzureKubernetesTokenEndpoint = "AZURE_KUBERNETES_TOKEN_ENDPOINT"
+)
+
+func parseAndValidateCustomTokenEndpoint(endpoint string) (*url.URL, error) {
```

Comment:
Do we really need to double-check the work of whatever internal component sets the environment variable?

Reply:
Do you mean SDK users i.e. application developers are responsible for setting `AZURE_KUBERNETES_TOKEN_ENDPOINT`?

Reply:
it's for those non-AKS users like [capz](https://capz.sigs.k8s.io/). In that case, the cluster owner will need to setup the token infrastructure then inject the settings like this.